src/main/java/com/autonomic/tmc/auth/exception/SdkServiceException.java

```diff
@@ -0,0 +1,40 @@
+/*-
+ * ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+ * TMC Auth SDK
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Copyright (C) 2016 - 2020 Autonomic, LLC
+ * ——————————————————————————————————————————————————————————————————————————————
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License
+ * ______________________________________________________________________________
+ */
+package com.autonomic.tmc.auth.exception;
+
+import static com.autonomic.tmc.auth.exception.ErrorSourceType.SERVICE;
+
+public class SdkServiceException extends BaseSdkException {
```

Comment:
What do you think about including the original exception?

We don't have to log it, but, if someone needs to do some serious debugging (like Kyle had to do), then they at least have access to it.

Reply:
Added a getMessage, with the errorCode and the error message.

Reply:
Added a `getErrorCause()` to see original exception and `getStatusCode` to see errorResponseCode.